### PR TITLE
Trm 23623 use new uniprot search config

### PR DESCRIPTION
**Pull request by sahmad-ebi**

### CODE REVIEW

File: uniprot-config/src/main/resources/uniprot-fields.json [hunk 1]
```diff
@@ -30,7 +57,8 @@
     "dataType": "string",
     "example":  "sample example",
     "fieldType": "field",
-    "description" : "this is a dummy description"
+    "description" : "this is a dummy description",
```
Comment (eddturner) [inline uniprot-config/src/main/resources/uniprot-fields.json]: should we remove all the dummy descriptions, or do you think it adds value?

File: uniprot-search/src/main/java/org/uniprot/store/search/domain2/UniProtKBSearchFields.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    @Override
+    public Set<SearchField> getSearchFields() {
+        if (this.searchFields == null) {
+
+            this.searchFields =
+                    this.configService.getAllFieldItems().stream()
+                            .filter(this::isSearchField)
+                            .map(fieldItem -> SearchFieldImpl.from(fieldItem))
```
Comment (eddturner) [inline uniprot-search/src/main/java/org/uniprot/store/search/domain2/UniProtKBSearchFields.java]: can use method ref

(the intellij sonarqube plugin is useful for spotting these suggestions. i use it)

File: uniprot-search/src/main/java/org/uniprot/store/search/domain2/UniProtKBSearchFields.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        return sortField;
+    }
+
+    @Override
+    public Set<SearchField> getSortFields() {
+        if (this.sortFields == null) {
+            this.sortFields =
+                    this.configService.getAllFieldItems().stream()
+                            .filter(this::isSortField)
+                            .map(fieldItem -> SearchFieldImpl.from(fieldItem))
```
Comment (eddturner) [inline uniprot-search/src/main/java/org/uniprot/store/search/domain2/UniProtKBSearchFields.java]: can use method ref

File: uniprot-config/src/main/resources/uniprot-fields.json (hunk 1 above)
Comment (sahmad-ebi) [reply uniprot-config/src/main/resources/uniprot-fields.json]: We need to add actual description. Can we leave them until i start consuming in swagger?

**Comment by sahmad-ebi** [comment]
> 1. Could we rename uniprot -> uniprotkb, e.g., in `UniProtSearchFieldConfiguration` be renamed `UniProtKBSearchFieldConfiguration`?
> 
> Same for UniProtSearchFieldConfiguration and any other classes recently created which I missed previously (sorry!).
> 
> 1. from slack:
> 
> > Shadab: Today at 10:10
> > We need thin UniProtSearchFieldConfiguration for each data source to initialise the right config file
> > Edward Turner  19 minutes ago
> > is the only difference going to be the names of the config files (fields + schema)? If so, code could be reused

Taken care of.

File: uniprot-config/src/main/resources/uniprot-fields.json (hunk 1 above)
Comment (eddturner) [reply uniprot-config/src/main/resources/uniprot-fields.json]: sure

**Pull request merged**